pkg/handler/eval_cache.go

```diff
@@ -70,13 +92,43 @@ func (ec *EvalCache) GetByTags(tags []string) []*entity.Flag {
 			}
 		}
 	}
+	return results
+}
 
-	values := make([]*entity.Flag, 0, len(results))
-	for _, f := range results {
-		values = append(values, f)
+func (ec *EvalCache) getByTagsALL(tags []string) map[uint]*entity.Flag {
+	results := map[uint]*entity.Flag{}
+	cache := ec.cache.Load().(*cacheContainer)
+
+	for i, t := range tags {
+		fSet, ok := cache.tagCache[t]
+		if !ok {
+			// clear as no flags
+			results = map[uint]*entity.Flag{}
+			continue
```

Comment:
if the tag is not there, I think should we just return `nil` here given the `ALL` operator.

Reply:
You're totally right

Reply:
But is it correct to return nil or should we return an empty list?